students/aravichander/lesson09/Mailroom_class.py

```
@@ -73,54 +110,57 @@ def letterofthanks(donorname,totaldonated):
 	f.close()
 	print("\nLetter of thanks has been saved to hard disk as text file under "+str(filename))
 
-def sumdonor(thankdonor):
-	return sum(Donors[thankdonor])
+# def sumdonor(thankdonor): #redundant - refactor to remove this
+# 	return sum(Donors[thankdonor])
 
-def searchdonors(thankdonor):
+def searchdonors(thankdonor,db):
 		try:
-			print(Donors[thankdonor])
-			totaldonated = sumdonor(thankdonor)
+			#print(db[thankdonor])
+			donor = db.get_donor(thankdonor)
+			totaldonated = donor.sum_donation
 			letterofthanks(thankdonor,totaldonated)
 		except KeyError:
 			print("\nThat donor doesn't exist")
 
 
-def newdonation():
+def newdonation(db):
 	newdonor = input("Input name of new donor\n")
 	while len(str.strip(newdonor)) == 0:
 		newdonor = input("Invalid input, enter name of new donor\n")
 	newdonor = str.strip(newdonor)
-	Donors[newdonor] = []
+	try:
+		newdonor_object = db.get_donor(newdonor)
+	except:	
```

never use  bare Except!

this would be a KeyError, yes?